doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,382 @@
+    such that subsequent requests can be served from cache.
+
+
+PCB dissemination
+^^^^^^^^^^^^^^^^^^^^^^^
+
+The IndexIdentifierMap and SupportedIndicesMap are included in a (unsigned) detachable extension in the PCBs for an AS.
+Hashes of these maps are maintained in a Signed AS Entry, such that the authenticity of these maps can be verified.
+If the maps are detached, they can be fetched from the control service of that AS and the received maps can be verified with the hashes.
+To ensure a consistent hash calculation, the key entries of these maps have to be sorted, such that they are accessed in a consistent order.
+
+Exposing policies to the end hosts
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The path combinator finds the most recent FABRID map per AS among the received segments and subsequently uses this map to find the FABRID
+policies that are available for each interface pair of hops.
+This results in a set of PolicyIdentifiers per hop, which can then be used by the application, such as with the usage of a
+specific ‘sequence’ parameter which incorporates the policies.
+Once the application has decided which policies to use, it can craft a FABRID HBH extension and include this as an option when sending
+the packet.
+
+DRKey
+^^^^^^
+
+FABRID uses DRKey for computing the Encrypted Policy Indices, the FABRID Hop Validation Fields and the Path Validator.
+The routers use the fast key derivation side, whereas the endhosts will use the slow side.
+
+Configuration
+--------------
+
+Control service
+^^^^^^^^^^^^^^^^^^
+
+To be able to use DRKey, one has to configure the control service setting "drkey.level1_db" and "drkey.secret_value_db".
+Additionally, since the border routers will fetch the secret value from the control service, the control service also has to
+add the internal IP address of all border routers of the local AS to the DRKey delegation list for FABRID.
+
+This could look like this::
+
+    [drkey.level1_db]
+    connection = "gen-cache/cs1-ff00_0_110-1.drkey-level1.db"
+
+    [drkey.secret_value_db]
+    connection = "gen-cache/cs1-ff00_0_110-1.drkey-secret.db"
+
+    [drkey.delegation]
+    FABRID = [ "fd00:f00d:cafe::7f00:11", "fd00:f00d:cafe::7f00:12", "fd00:f00d:cafe::7f00:13"]
+
+The FABRID policies are configured in the control service. TODO(jelte): add more details
+
+Border router
+^^^^^^^^^^^^^^^
+
+For a router to query the DRKey secret value from the control service, once has to enable this.
+
+This could look like this::
+
+    [router]
+    use_drkey = true
+
+Considerations for future work
+--------------------------------
+
+SCMP response
+^^^^^^^^^^^^^^^
+
+With the current implementation, the sending endhost is not being informed when his packet gets dropped due to a FABRID error.
+In the future the border routers might send an SCMP response if they encounter an error when processing FABRID which might
+help the sending endhost in figuring out why his packet does not arrive at its destination.
```

Comment:
Also briefly mention that we have to be careful not to introduce additional attack vectors, e.g., for volumetric DDoS attacks / reflection attacks etc.